ehrapy/api/_anndata_util.py

```diff
@@ -36,3 +36,66 @@ def get_column_values(adata: AnnData, indices: Union[int, List[int]]) -> np.ndar
         :class:`~numpy.ndarray` object containing the column values
     """
     return np.take(adata.X, indices, axis=1)
+
+
+def assert_encoded(adata: AnnData):
+    try:
+        assert "categoricals" in adata.uns_keys()
+    except AssertionError:
+        raise NotEncodedError from AssertionError("The AnnData object has not yet been encoded.")
+
+
+def get_numeric_vars(adata: AnnData) -> List[str]:
+    """Fetches the column names for numeric variables in X.
+
+    Args:
+        adata: :class:`~anndata.AnnData` object
+
+    Returns:
+        Set of column numeric column names
+    """
+
+    assert_encoded(adata)
+
+    return adata.uns["categoricals"]["not_categorical"]
+
+
+def set_numeric_vars(
+    adata: AnnData, values: np.ndarray, vars: Optional[List[str]] = None, copy: bool = False
+) -> Optional[AnnData]:
+    """Sets the column names for numeric variables in X.
+
+    Args:
+        adata: :class:`~anndata.AnnData` object
+        values: Matrix containing the replacement values
+        vars: List of names of the numeric variables to replace. If `None` they will be detected using ~ehrapy.pp.get_numeric_vars.
+        copy: Whether to return a copy with the normalized data.
+
+    Returns:
+        :class:`~anndata.AnnData` object with updated X
+    """
+
+    if vars is None:
+        vars = get_numeric_vars(adata)
+
+    if not np.issubdtype(values.dtype, np.number):
```

Comment:
Is it np.number or np.float32? The former would have been my intuition.

Reply:
StackOverflow told me to use `np.number` to check for any kind of numeric matrix but happy to change it.

Reply:
@Imipenem got an idea here? Is `np.number` just a superset of `np.float32`? Any issue for us?

Reply:
@Zethson, I do not think we will get into any trouble here. 
In fact, since line 78 enforces the AnnData object to be already encoded and our default AnnData type (for X of course) is _float32_, this should be fine. 
IIRC, we don't even have an option to change X to be anything else than _float32_, when encoding columns (might need to implement this to give user more control over their data).